### PR TITLE
Closes #578 | Add Dataloader ViCon

### LINKED ISSUE #578: Create dataset loader for ViCon
Dataloader name: `vicon/vicon.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?vicon

|     Dataset| vicon |
|-------------|---|
| Description | ViCon comprises pairs of synonyms and antonyms across word classes, thus offering data to distinguish between similarity and dissimilarity. |
| Subsets | - |
| Languages | vie |
| Tasks | Word Analogy |
| License | Creative Commons Attribution Non Commercial Share Alike 2.0 (cc-by-nc-sa-2.0) |
| Homepage | https://www.ims.uni-stuttgart.de/forschung/ressourcen/experiment-daten/vnese-sem-datasets/ |
| HF URL | - |
| Paper URL | https://aclanthology.org/N18-2032/ |

### PR DESCRIPTION
**Pull request by muhammadravi251001**
**Title**: Add Dataloader ViCon

**First line PR Message**: Closes https://github.com/SEACrowd/seacrowd-datahub/issues/578

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/{my_dataset}/{my_dataset}.py` (please use only lowercase and underscore for dataset folder naming, as mentioned in dataset issue) and its `__init__.py` within `{my_dataset}` folder.
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_LOCAL`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py` or `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py --subset_id {subset_name_without_source_or_seacrowd_suffix}`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: seacrowd/sea_datasets/vicon/vicon.py
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/vicon/vicon.py:90]: maybe use `POS_TAGS` or `CLASSES` here? Same comment with other instances of `TYPE` and `TYPES`

File: seacrowd/sea_datasets/vicon/vicon.py
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/vicon/vicon.py:187]:
```suggestion
```

Comment (muhammadravi251001) [reply seacrowd/sea_datasets/vicon/vicon.py]: Solved on https://github.com/SEACrowd/seacrowd-datahub/pull/607/commits/8e8bbfbd5c8aca0e0b0a75d996e58f5ef0a3cfda.

File: seacrowd/sea_datasets/vicon/vicon.py
Comment (muhammadravi251001) [reply seacrowd/sea_datasets/vicon/vicon.py]: Solved on https://github.com/SEACrowd/seacrowd-datahub/pull/607/commits/ad58e33b12fa66eed4907ba9152c90da52b4cd6a.

File: seacrowd/sea_datasets/vicon/vicon.py
Comment (muhammadravi251001) [reply seacrowd/sea_datasets/vicon/vicon.py:75]: I think the same with you, Sir.

I just don't think the task is supposed to be `WORD_ANALOGY`, I think the task is more suitable for `TEXTUAL_ENTAILMENT`, because it creates `pairs` schema, I don't think that `pairs_score` is suitable because the label is class-based (not Value with `float64` format).

But, from the issues https://github.com/SEACrowd/seacrowd-datahub/issues/578, the Task written is `WORD_ANALOGY`.

What do you think about this, Sir?

Comment (muhammadravi251001) [reply seacrowd/sea_datasets/vicon/vicon.py]: I've already changed the Task to `TEXTUAL_ENTAILMENT`, you can test it with these three commands, Sir.
```
python -m tests.test_seacrowd seacrowd/sea_datasets/vicon/vicon.py --subset_id vicon_noun
python -m tests.test_seacrowd seacrowd/sea_datasets/vicon/vicon.py --subset_id vicon_adj
python -m tests.test_seacrowd seacrowd/sea_datasets/vicon/vicon.py --subset_id vicon_verb
```
What do you think about that, Sir? Are we allowed to change Task from issues https://github.com/SEACrowd/seacrowd-datahub/issues/578?

Comment (yongzx) [reply seacrowd/sea_datasets/vicon/vicon.py]: My two cents here are that it seems like the confusion is due to the task schema type. Currently `WORD_ANALOGY` belongs to the text2text schema where this dataset (I've checked https://www.ims.uni-stuttgart.de/forschung/ressourcen/experiment-daten/vnese-sem-datasets/) seems to be classification of two paired strings.

Unless there's a change in the schema type for `WORD_ANALOGY`, I think that `TEXTUAL_ENTAILMENT` is the best alternative. Curious about @holylovenia and @SamuelCahyawijaya think too.

**Comment by yongzx** [comment]
```
python -m tests.test_seacrowd seacrowd/sea_datasets/vicon/vicon.py --subset_id vicon_noun
python -m tests.test_seacrowd seacrowd/sea_datasets/vicon/vicon.py --subset_id vicon_adj
python -m tests.test_seacrowd seacrowd/sea_datasets/vicon/vicon.py --subset_id vicon_verb
```
Thanks @muhammadravi251001! All the tests passed on my end.

**Comment by muhammadravi251001** [comment]
> ```
> python -m tests.test_seacrowd seacrowd/sea_datasets/vicon/vicon.py --subset_id vicon_noun
> python -m tests.test_seacrowd seacrowd/sea_datasets/vicon/vicon.py --subset_id vicon_adj
> python -m tests.test_seacrowd seacrowd/sea_datasets/vicon/vicon.py --subset_id vicon_verb
> ```
> Thanks @muhammadravi251001! All the tests passed on my end.

Alright, thanks for the info, Sir!

File: seacrowd/sea_datasets/vicon/vicon.py
Comment (holylovenia) [reply seacrowd/sea_datasets/vicon/vicon.py]: I agree with your decision to convert this to the `pairs` schema.

Which one do you suppose is a more suitable task for the dataset, `TEXTUAL_ENTAILMENT` or [`CONCEPT_ALIGNMENT_CLASSIFICATION` (e.g., id_am2ico)](https://github.com/SEACrowd/seacrowd-datahub/blob/0d07348ac976104cf28ee47f4cd93c9d02489998/seacrowd/sea_datasets/id_am2ico/id_am2ico.py#L71)? I'll follow whichever you'll decide on.

cc: @muhammadravi251001 @yongzx @ljvmiranda921

Comment (muhammadravi251001) [reply seacrowd/sea_datasets/vicon/vicon.py]: > I agree with your decision to convert this to the `pairs` schema.
> 
> Which one do you suppose is a more suitable task for the dataset, `TEXTUAL_ENTAILMENT` or [`CONCEPT_ALIGNMENT_CLASSIFICATION` (e.g., id_am2ico)](https://github.com/SEACrowd/seacrowd-datahub/blob/0d07348ac976104cf28ee47f4cd93c9d02489998/seacrowd/sea_datasets/id_am2ico/id_am2ico.py#L71)? I'll follow whichever you'll decide on.
> 
> cc: @muhammadravi251001 @yongzx @ljvmiranda921

I guess, because of the similarity of this task to the NLI task; NLI task labels are: `entailment`, `neutral`, and `contradiction`; and this task labels are: `synonym` and `antonym`, both concepts are just the same, **to find the relationship between `text_1` and `text_2`** whether is `synonym` (`entailment`) or `antonym` (`contradiction`).

We can assume that `synonym` is correlated to `entailment` and `antonym` correlates to `contradiction`.

So I chose `TEXTUAL_ENTAILMENT` (because NLI's task was it too).

File: seacrowd/sea_datasets/vicon/vicon.py
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/vicon/vicon.py:76]: Maybe a better description? Something like

```suggestion
# This task is more suitable for TEXTUAL_ENTAILMENT
# because the labels (antonym, synonym) roughly correlates to (contradiction, entailment)
```

**Review by ljvmiranda921 — APPROVED** [review]
Just a small suggestion and I think we're good 👍

**Comment by muhammadravi251001** [comment]
> Just a small suggestion and I think we're good 👍 

I've already commit the suggestion, Sir. Thanks!

**Comment by ljvmiranda921** [comment]
Just waiting for @yongzx 's approval and we can merge :)

**Comment by holylovenia** [comment]
> ```
> python -m tests.test_seacrowd seacrowd/sea_datasets/vicon/vicon.py --subset_id vicon_noun
> python -m tests.test_seacrowd seacrowd/sea_datasets/vicon/vicon.py --subset_id vicon_adj
> python -m tests.test_seacrowd seacrowd/sea_datasets/vicon/vicon.py --subset_id vicon_verb
> ```
> 
> Thanks @muhammadravi251001! All the tests passed on my end.

Hey @yongzx, I'm assuming you meant to approve this PR, thus I'm merging it.

**Pull request merged**
